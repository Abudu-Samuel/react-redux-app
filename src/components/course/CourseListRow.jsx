import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CourseListRow = ({ course, handleDelete }) => (
  <tr>
    <td><a href={course.watchHref}>Watch</a></td>
    <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
    <td>{course.authorId}</td>
    <td>{course.category}</td>
    <td>{course.length}</td>
    <td><button onClick={() => handleDelete(course)} className="btn-danger" type="button">Delete</button></td>
  </tr>
);

CourseListRow.propTypes = {
  course: PropTypes.object.isRequired
};

export default CourseListRow;
