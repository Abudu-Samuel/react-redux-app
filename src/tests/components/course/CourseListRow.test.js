/* eslint-disable */
import CourseListRow from '../../../components/course/CourseListRow';
import { MemoryRouter } from 'react-router-dom';

const props = {
  course: '',
  handleDelete: jest.fn()
};

const config = () => {
  return mount(
    <MemoryRouter>
      <CourseListRow {...props} />
    </MemoryRouter>
  );
};

let mountWrap;

describe('<CourseListRow />', () => {
  it('should render CourseListRow component properly', () => {
    mountWrap = config();
    expect(toJson(mountWrap)).toMatchSnapshot();
  });

  it('should delete course', () => {
    mountWrap = config();
    const buttonToSimulate = mountWrap.find('button').at(0);
    buttonToSimulate.simulate('click')
    expect(props.handleDelete).toBeCalled();
  });
});
