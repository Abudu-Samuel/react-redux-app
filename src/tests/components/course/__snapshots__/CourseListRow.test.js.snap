// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseListRow /> should render CourseListRow component properly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "g17lhw",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "g17lhw",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CourseListRow
      course=""
      handleDelete={[MockFunction]}
    >
      <tr>
        <td>
          <a>
            Watch
          </a>
        </td>
        <td>
          <Link
            replace={false}
            to="/course/undefined"
          >
            <a
              href="/course/undefined"
              onClick={[Function]}
            />
          </Link>
        </td>
        <td />
        <td />
        <td>
          0
        </td>
        <td>
          <button
            className="btn-danger"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
    </CourseListRow>
  </Router>
</MemoryRouter>
`;
